scalacheck-effect/src/main/scala/org/specs2/scalacheck/effect/ScalaCheckEffectProperty.scala

```diff
@@ -0,0 +1,1376 @@
+
+  def setSeed(seed: String): SelfType
+
+  def setVerbosity(v: Int): SelfType =
+    setParameters(parameters.setVerbosity(v))
+
+  def set(
+      minTestsOk: Int = parameters.minTestsOk,
+      minSize: Int = parameters.minSize,
+      maxDiscardRatio: Float = parameters.maxDiscardRatio,
+      maxSize: Int = parameters.maxSize,
+      workers: Int = parameters.workers,
+      callback: Test.TestCallback = parameters.testCallback,
+      loader: Option[ClassLoader] = parameters.loader
+  ): SelfType =
+    setParameters(
+      parameters.copy(
+        minTestsOk = minTestsOk,
+        minSize = minSize,
+        maxDiscardRatio = maxDiscardRatio,
+        maxSize = maxSize,
+        workers = workers,
+        testCallback = callback,
+        loader = loader
+      )
+    )
+
+  def display(
+      minTestsOk: Int = parameters.minTestsOk,
+      minSize: Int = parameters.minSize,
+      maxDiscardRatio: Float = parameters.maxDiscardRatio,
+      maxSize: Int = parameters.maxSize,
+      workers: Int = parameters.workers,
+      callback: Test.TestCallback = parameters.testCallback,
+      loader: Option[ClassLoader] = parameters.loader
+  ): SelfType =
+    setParameters(
+      parameters
+        .copy(
+          minTestsOk = minTestsOk,
+          minSize = minSize,
+          maxDiscardRatio = maxDiscardRatio,
+          maxSize = maxSize,
+          workers = workers,
+          testCallback = callback,
+          loader = loader
+        )
+        .setVerbosity(1)
+    )
+
+  def verbose: SelfType =
+    setVerbosity(1)
+
+  def setPrettyFreqMap(f: FreqMap[Set[Any]] => Pretty): SelfType
+
+  def prettyFreqMap(f: FreqMap[Set[Any]] => String): SelfType =
+    setPrettyFreqMap((fq: FreqMap[Set[Any]]) => Pretty(_ => f(fq)))
+
+object ScalaCheckEffectProperty:
+
+  given ScalaCheckEffectPropertyAsExecution[F[_]: MonadThrow, S <: ScalaCheckEffectProperty[F]](using
+      AsExecution[F[Result]]
+  ): AsExecution[S] with
+    def execute(s: =>S): Execution =
+      Execution.withEnvFlatten { env =>
+        AsResultPropF.check(s.propF, s.parameters.overrideWith(env.commandLine), s.prettyFreqMap)
+      }
+
+  def makePropF[F[_]: MonadThrow, T](f: T => PropF[F], shrink: Option[Shrink[T]], parameters: Parameters)(implicit
+      a: Arbitrary[T],
+      p: T => Pretty
+  ): PropF[F] =
+    shrink match
+      case Some(s) => PropF.forAllShrinkF(a.arbitrary, s.shrink)(f)
+      case _       => PropF.forAllNoShrinkF(f)
+
+/** A ScalaCheckEffectFunction adds the possibility to select various typeclass instances for a given property:
+  *
+  *   - Arbitrary to generate values
+  *   - Shrink to shrink counter-examples
+  *   - Show to display arguments in case of a counter-example
+  *   - Collector to collect values and provide a summary as string (to show frequencies for example)
+  */
+trait ScalaCheckEffectFunction[F[_]] extends ScalaCheckEffectProperty[F]:
+  def noShrink: SelfType
+
+  def context: Option[Context]
```

Comment:
I think we should remove this `context` stuff. I don't know why but it even got lost on the 5.x branch, `executeInContext` should have been used in `lazy val propFunction`. I'm going to fix in specs2 5.x but I'm thinking that it is better to keep the API simple here

Reply:
I'm 💯  for simplifications!